src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js

```diff
@@ -294,23 +300,27 @@
         },
         _createSourceTitleLine: function(layer){
             return '<li class="ebene' + layer.level + ' ' + this.sourceTitle + (!layer.childrenLegend ? ' notshow'
-                    : '') + ' title" data-sourceid="' + layer.sourceId + '" data-id="' + layer.id + '">' + layer.title + '</li>';
+                : '') + ' title" data-sourceid="' + layer.sourceId + '" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createNodeTitleLine: function(layer){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.grouppedTitle + (!layer.childrenLegend
-                    ? ' notshow'
-                    : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
+                ? ' notshow'
+                : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createTitleLine: function(layer, hide){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.layerTitle + ' ' + (hide
-                    ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
+                ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createImageLine: function(layer){
-            return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
+
+            if(layer.visible == " notvisible")
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '" data-gisurl="' + layer.legend.url + '"></li>';
+            else
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
```

Comment:
I agree with this point. Also in my opinion we should not write markup directly in the js code. Instead we should use jQuery, so it will be easier to modifiy the structure in the future.